### ISSUE TITLE
False negative typecheck error after using Omit with union type.

### ISSUE THREAD

**Issue by Akiyamka** [microsoft/TypeScript#53696]
# Bug Report
Omitting one property from union type broke typecheck of other properties in union.
Typescript completely loose property if it abset at least in one of types from union after Omit helper,
"Object literal may only specify known properties, and 'property_that_actually_exist' does not exist in type 'Omit<AnyLayer, "id">'" 
### 🔎 Search Terms

omit breaks union

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about 

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5JA)

### 💻 Code

https://www.typescriptlang.org/play?ts=5.0.3#code/FASwdgLgpgTgZgQwMZQAQDEQBssBkECesqA3sKhaiACYBcqAzhDOAOYDc5lEBADlPQDkcbFkGdKqXjAD2-GDwD6EABYIIi5BACuCHAUVQAHiCYB+ekxZgOwAL7Bgx3jIVVIsRClQBhbUxkAW3wiGABJD3hkNDJJGktmNgluPgFUQSR-CCDxe0dnVwhUHn5UAEEwAhDiAF4uCgAfDFFqmHrUJr8A4MJYCOgolE5HAHoAKlQAFSgmVCQEBjQAdxBVVCDV1DGR4CQZMFmsXpgAZR4sGdQa8pgYQgAeAHlA1fuKquOAGnSaQQA+P4ACgAlJwjqEzgQLgwAHS8fwqQGxFL8IQiHCCT7taRyWBKVTqTRIHR6LAGYymCBCBgybQwFCY+zA0YTaazeaLVArVS0oobIrbXb7Q7HSHQgBMVxudwIb0qrSBoOA4NgYpm4rhCKR7Xi6TgMhkjMkJTSwlERsoOPk+LUGi0un0hhMTGptPpUEZdmBQA

### 🙁 Actual behavior
false negative typecheck error after using Omit with union type.

### 🙂 Expected behavior
Omit not broke union typecheck

**Comment by Akiyamka** [comment]
For those who encounter the same problem, a temporary solution looks like this:
```ts
export type FixedOmit<T, U> = T extends any ? Pick<T, Exclude<keyof T, U>> : never
```
[Playground link](https://www.typescriptlang.org/play?ts=5.0.3#code/FASwdgLgpgTgZgQwMZQAQDEQBssBkECesqA3sKhaiACYBcqAzhDOAOYDc5lEBADlPQDkcbFkGdKqXjAD2-GDwD6EABYIIi5BACuCHAUVQAHiCYB+ekxZgOwAL7Bgx3jIVVIsRClQBhbUxkAW3wiGABJD3hkNDJJGktmNgluPgFUQSR-CCDxe0dnVwhUHn5UAEEwAhDiAF4uCgAfDFFqmHrUJr8A4MJYCOgolE5HAHoAKlQAFSgmVCQEBjQAdxBVVCDV1DGR4CQZMFmsXpgAZR4sGdQa8pgYQgAeAHlA1fuKquOAGnSaQQA+P4ACgAlJwjqEzgQLgwAHS8fwqQGxFL8IQiHCCT7taRyWBKVTqTRIHR6LAGYymCBCBgybQwFCY+zA0YTaazeaLVArVS0oobIrbXb7Q7HSHQgBMVxudwIb0qrSBoOA4NgYpm4rhCKR7Xi6TgMhkjMkJTSwlERsoOPk+LUGi0un0hhMTGptPpUEZdmZwHGUxmRQ5y1WKi5rgQsm0YGoWx2JuaRig1Ger0m3wAqn8pZNUMZoFGGKgEJVUGZUAAFEBIADW91TqAAokYkFhtNQoPcq1ACDI4FN0wDUPQwFAAG6wYZ7A5FFWnc4zADMUrKtwemATSZeEDlH1C30EvwBILBornDHnmoYiORFBNaPNWMkVrxBgJduJDrJTsprrpDKxXqAA)
